code/modules/unit_tests/unit_test.dm

```diff
@@ -33,6 +33,9 @@ GLOBAL_VAR_INIT(focused_tests, focused_tests())
 	//Bit of metadata for the future maybe
 	var/list/procs_tested
 
+	/// If there are any failed tests prior to this test, do not run it
+	var/only_run_if_no_failures = FALSE
```

Comment:
This doesn't do what you want.

Its not good enough to just add a bool flag.

There was a *reason* why i suggested grouping unit tests into stages or buckets.

You can't do shit like this without sorting, and ensuring C&D runs *LAST* and if you are gonna sort, then there is no reason to not make it expandable by making it a stage number or aligning it to one of a given set of "passes" thru the unit test queue, and skipping all future passes/stages/whatever if any in the current stage fails.

Making it expandable like this makes testing new/modified unit tests EVEN FASTER, now you just temporarily put it in stage `0` a normally empty stage, so it runs first and doesn't waste CI time running the later tests once it fails.